lib/beacon/live_admin/live/error_page_editor_live/index.ex

```diff
@@ -0,0 +1,284 @@
+    {:noreply, assign(socket, show_delete_modal: false)}
+  end
+
+  def handle_event("stay_here", _params, socket) do
+    {:noreply, assign(socket, show_nav_modal: false, confirm_nav_path: nil)}
+  end
+
+  def handle_event("discard_changes", _params, socket) do
+    {:noreply, push_redirect(socket, to: socket.assigns.confirm_nav_path)}
+  end
+
+  def handle_event("cancel_create", _params, socket) do
+    {:noreply, assign(socket, show_create_modal: false)}
+  end
+
+  defp assign_selected(socket, nil) do
+    case socket.assigns.error_pages do
+      [] -> assign(socket, selected: nil, changed_template: "")
+      [hd | _] -> assign(socket, selected: hd, changed_template: hd.template)
+    end
+  end
+
+  defp assign_selected(socket, status) when is_binary(status) do
+    assign_selected(socket, String.to_integer(status))
+  end
+
+  defp assign_selected(socket, status) when is_integer(status) do
+    selected = Enum.find(socket.assigns.error_pages, &(&1.status == status))
+    assign(socket, selected: selected, changed_template: selected.template)
+  end
+
+  defp assign_form(socket) do
+    form =
+      case socket.assigns do
+        %{selected: nil} ->
+          nil
+
+        %{selected: selected, beacon_page: %{site: site}} ->
+          site
+          |> Content.change_error_page(selected)
+          |> to_form()
+      end
+
+    assign(socket, form: form)
+  end
+
+  defp assign_error_page_update(socket, updated_error_page) do
+    %{id: error_page_id} = updated_error_page
+
+    error_pages =
+      Enum.map(socket.assigns.error_pages, fn
+        %{id: ^error_page_id} -> updated_error_page
+        other -> other
+      end)
+
+    assign(socket, error_pages: error_pages)
+  end
+
+  def render(assigns) do
+    ~H"""
+    <div>
+      <.header>
+        <%= @page_title %>
+        <:actions>
+          <.button phx-disable-with="Saving..." form="error-page-form" class="uppercase">Save Changes</.button>
+        </:actions>
+        <:actions>
+          <.button type="button" id="delete-error-page-button" phx-click="delete" phx-disable-with="Deleting..." class="uppercase">Delete</.button>
+        </:actions>
+      </.header>
+
+      <.modal :if={@show_nav_modal} id="confirm-nav" on_cancel={JS.push("stay_here")} show>
+        <p>You've made unsaved changes to this error page!</p>
+        <p>Navigating to another error page without saving will cause these changes to be lost.</p>
+        <.button type="button" phx-click="stay_here">
+          Stay here
+        </.button>
+        <.button type="button" phx-click="discard_changes">
+          Discard changes
+        </.button>
+      </.modal>
+
+      <.modal :if={@show_create_modal} id="create-modal" on_cancel={JS.push("cancel_create")} show>
+        <.simple_form :let={f} for={@create_form} id="create-form" phx-submit="save_new">
+          <.input field={f[:status]} type="number" label="Status code for new error page:" />
```

Comment:
@leandrocp Originally in my first draft, I was limiting the error pages to only standard error status codes.  Changeset validation, etc.  But then I reverted this because I noticed that Phoenix/Plug actually allow non-standard status codes to be used (e.g. raise an error with status code 998, then handle the error with `render("998.html"...`), and I wasn't sure we wanted to lose this feature in Beacon.  Do you think the benefit is worth it?

Reply:
> Phoenix/Plug actually allow non-standard status codes to be used

That doesn't seems to be a very used feature - https://github.com/search?q=%2Fconfig+%3Aplug.*%3Astatuses%2F+lang%3Aelixir&type=code, it seems to solve exceptions to the usual cases. So between that or proving better UX and [consistency in Beacon](https://github.com/BeaconCMS/beacon/pull/331#discussion_r1306112415) I'd prefer to limit the status codes for now and only allow extension _if_ necessary.